src/snippet/mod.rs

```diff
@@ -186,6 +186,28 @@ fn select_best_fragment_combination(fragments: &[FragmentCandidate], text: &str)
     }
 }
 
+/// Returns ranges that are collapsed into non-overlapped ranges.
+fn collapse_overlapped_ranges(ranges: &Vec<Range<usize>>) -> Vec<Range<usize>> {
+    if ranges.is_empty() {
+        return Vec::new();
+    }
+
+    let mut result = Vec::new();
+    let mut current = ranges[0].clone();
```

Comment:
I am not sure but maybe it would be nicer to iterate only once, e.g.

```rust
let mut result = Vec::new();

let mut ranges = ranges.iter();

let mut current = match ranges.next() {
  Some(range) => range.clone(),
  None => return result,
};

for range in ranges {
  ...

Reply:
Sounds good. I changed so. Thanks.